client/state/jetpack-connect/reducer.js

```diff
@@ -118,7 +118,7 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			return Object.assign( {}, state, { autoAuthorize: false } );
 		case SERIALIZE:
 		case DESERIALIZE:
-			return {};
+			return state;
```

Comment:
I've already been experiencing some weird side effects that I believe are related to this.

For example: If I authorize a user, and then shortly after unlink and try to reauthorize again, I get something that looks like this:

![screen shot 2016-05-17 at 2 56 43 pm](https://cloud.githubusercontent.com/assets/1126811/15337110/a19cd62a-1c3f-11e6-9170-59c199a23a8a.png)

Note, the disabled button and the "Returning to your site" text. This is shown immediately to me after I try to reauthorize.

I am able to fix the issue but clearing the `calypso_store` in indexed DB and attempting to reauthorize, which leads me to believe it's due to stale state.